airflow/providers/amazon/aws/operators/aws_lambda.py

```diff
@@ -0,0 +1,121 @@
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import json
+from typing import TYPE_CHECKING, Optional, Sequence
+
+from airflow.models import BaseOperator
+from airflow.providers.amazon.aws.hooks.lambda_function import LambdaHook
+
+if TYPE_CHECKING:
+    from airflow.utils.context import Context
+
+
+class AwsLambdaInvokeFunctionOperator(BaseOperator):
+    """
+    Invokes an AWS Lambda function.
+    You can invoke a function synchronously (and wait for the response),
+    or asynchronously.
+    To invoke a function asynchronously,
+    set InvocationType to `Event`.
+
+    :param function_name: The name of the AWS Lambda function, version, or alias.
+    :param payload: The JSON string that you want to provide to your Lambda function as input.
+    :param log_type: Set to Tail to include the execution log in the response. Otherwise, set to "None".
+    :param qualifier: Specify a version or alias to invoke a published version of the function.
+    :param aws_conn_id: The AWS connection ID to use
+    """
+
+    template_fields: Sequence[str] = ('function_name', 'payload', 'qualifier', 'invocation_type')
+    ui_color = '#ff7300'
+
+    def __init__(
+        self,
+        *,
+        function_name: str,
+        log_type: Optional[str] = None,
+        qualifier: Optional[str] = None,
+        invocation_type: Optional[str] = None,
+        client_context: Optional[str] = None,
+        payload: Optional[str] = None,
+        aws_conn_id='aws_default',
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.FunctionName = function_name
+        self.Payload = payload
+        self.LogType = log_type
+        self.Qualifier = qualifier
+        self.InvocationType = invocation_type
+        self.ClientContext = client_context
+        self.aws_conn_id = aws_conn_id
+
+    def execute(self, context: 'Context'):
+        """
+        Invokes the target AWS Lambda function from Airflow.
+
+        :return: The response payload from the function, or an error object.
+        """
+        hook = LambdaHook(aws_conn_id=self.aws_conn_id)
+        success_status_codes = [200, 202, 204]
+        self.log.info("Invoking AWS Lambda function: %s with payload: %s", self.FunctionName, self.Payload)
+        try:
+            response = hook.invoke_lambda(
+                function_name=self.FunctionName,
+                **{
+                    key: self.__dict__[key]
```

Comment:
This should be refactored a bit and the `kwargs` extraction can be put outside of the `try` handler.

Reply:
@malthe, I understand. This has been refactored. Kindly review and resolve the conversation if the changes have addressed the comment.